packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,459 @@
+import 'dart:math' as math;
+
+import 'package:collection/collection.dart' as collection;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+/// Base class for delegate providing extent information for items in a list.
+abstract class ExtentDelegate {
+  /// Optional callback to execute after the layout of the extents is modified.
+  VoidCallback onLayoutDirty;
+
+  int get length;
+
+  /// The main-axis extent of each item.
+  double getItemExtent(int index);
+
+  /// The layout offset for the child with the given index.
+  double indexToLayoutOffset(int index);
+
+  /// The minimum child index that is visible at the given scroll offset.
+  ///
+  /// Implementations should take no more than O(log n) time.
+  int getMinChildIndexForScrollOffset(double scrollOffset);
+
+  /// The maximum child index that is visible at the given scroll offset.
+  ///
+  /// Implementations should take no more than O(log n) time.
+  /// Surprisingly, getMaxChildIndexForScrollOffset should be 1 less than
+  /// getMinChildIndexForScrollOffset if the scrollOffset is right at the
+  /// boundary between two items.
+  int getMaxChildIndexForScrollOffset(double scrollOffset);
+
+  @mustCallSuper
+  void recompute() {
+    if (onLayoutDirty != null) {
+      onLayoutDirty();
+    }
+  }
+}
+
+/// [ExtentDelegate]  implementation for the case where sizes for each
+/// item are known but absolute positions are not known.
+class FixedExtentDelegate extends ExtentDelegate {
+  FixedExtentDelegate({
+    @required this.computeExtent,
+    @required this.computeLength,
+  }) {
+    recompute();
+  }
+
+  List<double> _totalExtent;
+
+  @override
+  int get length => _totalExtent.length - 1;
+
+  final double Function(int index) computeExtent;
+  final int Function() computeLength;
+
+  @override
+  void recompute() {
+    final length = computeLength();
+    _totalExtent = List(length + 1);
+    double total = 0;
+    _totalExtent[0] = 0;
```

Comment:
Why are we setting this?

Reply:
commented. We should remove once we have NNBD